crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -323,6 +323,13 @@ async fn payment_response_update_tracker<F: Clone, T>(
                     .change_context(errors::ApiErrorResponse::InternalServerError)
                     .attach_printable("Could not parse the connector response")?;
 
+                // incase of success, update error code and error message
+                let error_status = if router_data.status == enums::AttemptStatus::Charged {
+                    Some(String::default())
```

Comment:
We should update null in DB right? Here it is updated with empty string.

Use Some(None) to update a value as NULL in Database, In ResponseUpdate for those fields use Option<Option<String>> as the type

Reply:
If we need to update as null we need to do as null only.

Empty string and null or not equal always